webknossos-datastore/app/com/scalableminds/webknossos/datastore/datareaders/zarr3/Zarr3Array.scala

```diff
@@ -122,8 +119,22 @@ class Zarr3Array(vaultPath: VaultPath,
       parsed = parseShardIndex(shardIndexRaw)
     } yield parsed
 
+  private lazy val shardIndexChecksumLength =
```

Comment:
I guess this works for most datasets, but not for all. Although, it doesn't make a lot of sense, it is spec-compliant to have multiple crc32c checksums. In the future, there might be other codecs that change the size of the shard index.

A more correct way would be to compute the size of the index based on the entire index_codecs pipeline. zarrita uses for that the [`compute_encoded_size` methods](https://github.com/scalableminds/zarrita/blob/async/zarrita/codecs.py#L545).

Reply:
At least, there should be some assertions that only supported index configurations are usable.

Reply:
#7574